tests/unit/parsing/test_inference_cache_flow.py

```
@@ -0,0 +1,115 @@
+            self.calls.append((query, kwargs))
+
+    class FakeDriver:
+        def __init__(self):
+            self.session_obj = FakeSession()
+
+        def session(self):
+            return self.session_obj
+
+    class FakeProjectManager:
+        def get_project_from_db_by_id_sync(self, repo_id):
+            return {"repo_path": "/tmp/repo"}
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/repo"

(S108)